### PR TITLE
Add zulu

### PR DESCRIPTION
**Pull request by ignatovich-artem**
add zulu https://github.com/lean-delivery/ansible-role-java/issues/75
fix issue https://github.com/lean-delivery/ansible-role-java/issues/78
some optimizations

### CODE REVIEW

File: tasks/Linux/fetch/s3.yml [hunk 1]
```diff
@@ -12,6 +12,6 @@
   retries: 5
   delay: 2
 
-- name: Set downloaded artifact var
+- name: Set downloaded artifact vars
   set_fact:
-    java_artifact_s3: '{{ download_path }}/{{ transport_s3_path|basename }}'
+    file_downloaded: {"dest": "{{ download_path }}/{{ transport_s3_path | basename }}"}
```
Comment (tgadiev) [inline tasks/Linux/fetch/s3.yml]:
```suggestion
    file_downloaded:
      dest: '{{ download_path }}/{{ transport_s3_path | basename }}'
```

File: tasks/Win32NT/fetch/openjdk-fallback.yml [hunk 2]
```diff
@@ -53,4 +53,4 @@
 
 - name: Set downloaded artifact vars
   set_fact:
-    java_artifact_fallback: '{{ artifact.stat.path }}'
+    file_downloaded: {"dest": "{{ artifact.stat.path }}"}
```
Comment (tgadiev) [inline tasks/Win32NT/fetch/openjdk-fallback.yml]:
```suggestion
    file_downloaded:
      dest: '{{ artifact.stat.path }}'
```

File: tasks/Win32NT/fetch/zulu-fallback.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    win_stat:
+      path: '{{ download_path }}\{{ artifact_basename }}'
+      get_checksum: true
+      checksum_algorithm: '{{ checksum_alg }}'
+    register: artifact
+
+  - name: Download with checksum validation
+    include_tasks: fetch_checksum.yml
+    when: |
+      not artifact.stat.exists
```
Comment (tgadiev) [inline tasks/Win32NT/fetch/zulu-fallback.yml]:
```suggestion
      not artifact.stat.exists | bool
```

File: tasks/Win32NT/fetch/zulu-fallback.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      or artifact.stat.checksum != checksum[0]
+    retries: 15
+    delay: 2
+    until: artifact.stat.checksum == checksum[0]
+
+  - name: Set downloaded artifact vars
+    set_fact:
+      file_downloaded: {"dest": "{{ artifact.stat.path }}"}
```
Comment (tgadiev) [inline tasks/Win32NT/fetch/zulu-fallback.yml]:
```suggestion
      file_downloaded:
        dest: '{{ artifact.stat.path }}'
```

File: tasks/Win32NT/install/zulu_tarball.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      recurse: false
+      file_type: directory
+    register: java_temp_folder
+
+  - name: Copy from temporary directory
+    win_copy:
+      src: '{{ java_temp_folder.files | map(attribute="path") | list | last }}\'
+      dest: '{{ java_path }}\{{ java_folder }}'
+      remote_src: true
+  when: not java_folder_bin.stat.exists
```
Comment (tgadiev) [inline tasks/Win32NT/install/zulu_tarball.yml]:
```suggestion
  when: not java_folder_bin.stat.exists | bool
```

File: tasks/Linux/fetch/s3.yml (hunk 1 above)
Comment (ignatovich-artem) [reply tasks/Linux/fetch/s3.yml]: Fixed

File: tasks/Win32NT/fetch/openjdk-fallback.yml (hunk 2 above)
Comment (ignatovich-artem) [reply tasks/Win32NT/fetch/openjdk-fallback.yml]: Fixed

File: tasks/Win32NT/fetch/zulu-fallback.yml (hunk 3 above)
Comment (ignatovich-artem) [reply tasks/Win32NT/fetch/zulu-fallback.yml]: Fixed

File: tasks/Win32NT/fetch/zulu-fallback.yml (hunk 4 above)
Comment (ignatovich-artem) [reply tasks/Win32NT/fetch/zulu-fallback.yml]: Fixed

File: tasks/Win32NT/install/zulu_tarball.yml (hunk 5 above)
Comment (ignatovich-artem) [reply tasks/Win32NT/install/zulu_tarball.yml]: Fixed

File: tasks/Win32NT/install/zulu_tarball.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+---
+- name: Check that the java_folder exists
+  win_stat:
+    path: '{{ java_path }}\{{ java_folder }}/bin'
```
Comment (tgadiev) [inline tasks/Win32NT/install/zulu_tarball.yml]:
```suggestion
    path: '{{ java_path }}\{{ java_folder }}\bin'
```

**Pull request merged**